### DISCUSSION TITLE
Docker JSON logs for mastodon containers are getting out of control; Mastodon documentation is silent on that

### DISCUSSION THREAD

**Discussion by penartur** [mastodon/mastodon#27911] [General]
### Expected behaviour

Mastodon works correctly for a long time, without overflowing my disk.

### Actual behaviour

In docker container directory, there is a JSON log file with the size of 14GB; there is no free disk space left, and Mastodon instance is unavailable. You have to delete this file manually from rescue console, and only after that will Mastodon instance become available again.

### Steps to reproduce the problem

Leave a federated Mastodon instance running for a long time (month+) unattended.

### Specifications

Mastodon version: v2.9.2.

According to Google, there are some Docker configuration settings that are supposed to stop this from happen (e.g. [this one](https://stackoverflow.com/a/42510314) that limits log size to manageable three files up to 10MB each). However, I'm not a Docker expert, and I have no idea where to put them.

Mastodon is supposed to be easy to install / support for people who are not experts in Linux administration or in Docker; however, Mastodon documentation does not feature any mention of that, too. I still don't understand how do I correctly prevent that problem from happening again; and I must be not the only one.

**Comment by Gargron** [comment]
You can adjust the Rails log level with `RAILS_LOG_LEVEL` environment variable which can vary from `debug` (most noisy) to `error` (least noisy). The default value is `info`, a step above `debug`, which makes information about requests and processed Sidekiq jobs visible.

This is an okay default because most installations aren't active enough to make this a problem, and journald takes care of log rotation. With Docker, this is clearly different.

Unfortunately, we've found that shooting oneself in the foot is easier with Docker, so Docker is no longer the recommended method of installation and has been removed from our documentation. It will continue to be supported, but it makes more sense for advanced users who use container-based infrastructure.
